### ISSUE TITLE
Failed to call method "__get__" for descriptor class "property"

### ISSUE THREAD

**Issue by medihack** [microsoft/pyright#6456]
**Describe the bug**
The below code gives me a type error with pyright when accessing the property. This happened the first time with pyright v1.335, but is still present with v1.336. There is no error with v1.334.

Error message: `Cannot access member "modality" for type "QueryDataset"
  Failed to call method "__get__" for descriptor class "property"`

**Code or Screenshots**
```python
class BaseDataset:
    def __init__(self):
        self._modality = "MR"

    @property
    def modality(self) -> str:
        return self._modality


class QueryDataset(BaseDataset):
    @BaseDataset.modality.setter
    def modality(self, value: str) -> None:
        self._modality = value


ds = QueryDataset()
x = ds.modality # <-- type error
```

**VS Code extension or command-line**
Pyright VS Code extension v1.1.336

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.337, which I just published. It will also be included in a future release of pylance.
